### PR TITLE
HrMethods

### PR DESCRIPTION
**Pull request by Yanetta**

### CODE REVIEW

File: src/main/java/oneMoreRest/MateGroupServiceImpl.java [hunk 1]
```diff
@@ -49,59 +51,97 @@ public Response addStudents(@PathParam("groupId") int groupId, Person person) {
...
+            mateGroup.getHumanResources().add(newHR);
+            return Response.status(Status.ACCEPTED).entity(mateGroup).type(MediaType.APPLICATION_JSON).build();
+        }
+        return Response.status(Status.NOT_FOUND).build();
+    }
+
+    @Override
+    @DELETE
+    @Path("/hrs/{name}")
+    @Consumes(MediaType.APPLICATION_JSON)
```
Comment (Taniann) [inline src/main/java/oneMoreRest/MateGroupServiceImpl.java]: Annotation @Consumes is redundant in this case

File: src/main/java/oneMoreRest/MateGroupServiceImpl.java [hunk 2]
```diff
@@ -49,59 +51,97 @@ public Response addStudents(@PathParam("groupId") int groupId, Person person) {
...
-          }
+    @Override
+    @GET
+    @Path("/hrs/{name}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getCertainHr(@PathParam("name") String name, @PathParam("groupId") int groupId) {
+        MateGroup mateGroup = mateGroups.get(groupId);
+        if (mateGroup != null) {
+            return Response.status(Status.OK).entity(mateGroup.getHumanResources().stream().filter(as -> as.getName().equals(name)).findAny().get()).type(MediaType.APPLICATION_JSON).build();
```
Comment (Taniann) [inline src/main/java/oneMoreRest/MateGroupServiceImpl.java]: if student with passed name is not exist we will get exception

File: src/main/java/oneMoreRest/MateGroupServiceImpl.java [hunk 3]
```diff
@@ -49,59 +51,97 @@ public Response addStudents(@PathParam("groupId") int groupId, Person person) {
...
+            mateGroup.getHumanResources().removeIf(hr -> hr.getName().equals(name));
+            return Response.status(Status.ACCEPTED).entity(mateGroup).type(MediaType.APPLICATION_JSON).build();
+        }
+        return Response.status(Status.NOT_FOUND).build();
+    }
+
+    @Override
+    @PUT
+    @Path("/hrs/{startWorkYear}")
+    @Consumes(MediaType.APPLICATION_JSON)
```
Comment (Taniann) [inline src/main/java/oneMoreRest/MateGroupServiceImpl.java]: @Consumes - same as above

File: src/main/java/oneMoreRest/MateGroupServiceImpl.java [hunk 4]
```diff
@@ -49,59 +51,97 @@ public Response addStudents(@PathParam("groupId") int groupId, Person person) {
...
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response changeHR(@PathParam("startWorkYear") int startWorkYear, @PathParam("groupId") int groupId, @QueryParam("newStartWorkYear") int newStartWorkYear) {
```
Comment (Taniann) [inline src/main/java/oneMoreRest/MateGroupServiceImpl.java:111]: maybe better to change hr's startWorkYear by name or surname. but in our case it's not so important

**Review by Taniann — APPROVED** [review]
good. the main task is done

File: src/main/java/oneMoreRest/MateGroupServiceImpl.java [hunk 5]
```diff
@@ -49,59 +51,97 @@ public Response addStudents(@PathParam("groupId") int groupId, Person person) {
...
+        MateGroup mateGroup = mateGroups.get(groupId);
+        if (mateGroup != null) {
+            mateGroup.getStudents().stream().filter(as -> as.getSurname().equals(surname)).forEach(as -> as.setName(newname));
+            return Response.status(Status.ACCEPTED).type(MediaType.APPLICATION_JSON).build();
+        }
+        return Response.status(Status.NOT_FOUND).build();
+    }
+
+    @Override
+    @PUT
```
Comment (SerhiiPasko) [inline src/main/java/oneMoreRest/MateGroupServiceImpl.java]: Post better

**Review by SerhiiPasko — APPROVED** [review]
Read comments

**Pull request merged**